File: commons/crypt/src/test/java/com/cedarsoft/crypt/ExpiredCertTest.java
package com.cedarsoft.crypt;

import java.nio.charset.StandardCharsets;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.*;

/**
 */
public class ExpiredCertTest {

  private X509Support x509Support;

  @BeforeEach
  public void setUp() throws Exception {
    x509Support = new X509Support(getClass().getResource("/test_expired.crt"), getClass().getResource("/test_expired.der"));
  }

  @Test
  public void testDate() throws Exception {
    byte[] plainText = "Hello World".getBytes(StandardCharsets.UTF_8);
    Signature signature = x509Support.sign(plainText);

    Assertions.assertThat(x509Support.verifySignature(plainText, signature)).isTrue();
  }
}

File: commons/crypt/src/test/kotlin/com/cedarsoft/crypt/ExpiredCertTest.kt
package com.cedarsoft.crypt

import com.cedarsoft.common.resources.getResourceSafe
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.nio.charset.StandardCharsets

/**
 */
class ExpiredCertTest {
  private var x509Support: X509Support? = null

  @BeforeEach
  @Throws(Exception::class)
  fun setUp() {
    x509Support = X509Support(javaClass.getResourceSafe("/test_expired.crt"), javaClass.getResourceSafe("/test_expired.der"))
  }

  @Test
  @Throws(Exception::class)
  fun testDate() {
    val plainText = "Hello World".toByteArray(StandardCharsets.UTF_8)
    val signature = x509Support!!.sign(plainText)
    Assertions.assertThat(x509Support!!.verifySignature(plainText, signature)).isTrue
  }
}
